pkg/api/types.go

```diff
@@ -195,6 +195,119 @@ type VolumeSource struct {
 	NFS *NFSVolumeSource `json:"nfs"`
 }
 
+// Similar to VolumeSource but meant for the administrator who creates PVs.
+// Exactly one of its members must be set.
+type PersistentVolumeSource struct {
```

Comment:
That makes sense.  I just thought I'd note the apparent duplication.  Happy go with whatever the most expedient solution is for now, as it doesn't seem to restrict us much in future.